scripts/mail-alarm

```diff
@@ -730,6 +707,7 @@ def main():
 
     finally:
         os.unlink(fname)
+        session._logout()
```

Comment:
This leaks a session if `main()` returns before the `try` block is reached. I think the try/finally should be at a higher scope, for example below around the invocation of mail itself, or perhaps by using an at exit handler.

Reply:
Right now `session` is a global variable. I would remove it from the global scope, make it local in `main()` and add the session as a parameter to all the functions that use it. This will allow for `session` and the temporary file to be cleaned up with a context manager using `with` construction.

Documentation on how to define a context manager: https://docs.python.org/2/library/contextlib.html

If you need an example, here's one that changes the working directory and goes back the execution leave the `with block`: https://github.com/common-workflow-language/cwltool/blob/master/tests/util.py#L103 and how it's used:  https://github.com/common-workflow-language/cwltool/blob/beab66d649dd3ee82a013322a5e830875e8556ba/tests/test_fetch.py#L86